src/cpu/x64/jit_uni_reorder.cpp

```diff
@@ -902,6 +904,341 @@ struct jit_uni_reorder_kernel_f32 : public kernel_t, public jit_generator {
     Zmm bf16_emu_reserv_4 = Zmm(19);
 };
 
+// Seperate class for no unroll/threading burden
+struct jit_single_blk_kernel : public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(jit_single_blk_kernel)
+    static bool applicable(const prb_t &p) {
+        using namespace data_type;
+
+        bool ok = true && p.ndims >= 2 && mayiuse(avx2)
+                && p.scale_type == scale_type_t::NONE
+                && utils::one_of(p.itype, f32 /*, bf16, s32, s8, u8*/)
+                && utils::one_of(p.otype, f32 /*, bf16, s32, s8, u8*/)
+                && IMPLICATION(
+                        p.itype == bf16, utils::one_of(p.otype, f32, bf16))
+                && IMPLICATION(
+                        p.otype == bf16, utils::one_of(p.itype, f32, bf16))
+                && utils::everyone_is(0, p.ioff, p.ooff) /* do we need this? */
+                && utils::one_of(p.beta, 0.f /*, 1.f*/) /* anything else? */
+                && IMPLICATION((p.itype == bf16 || p.otype == bf16),
+                        mayiuse(avx512_core));
+        if (!ok) return false;
+
+        int64_t a = p.nodes[0].n;
+        auto b = p.nodes[0].is;
+        auto c = p.nodes[0].os;
+        int64_t d = p.nodes[1].n;
+        auto e = p.nodes[1].is;
+        auto f = p.nodes[1].os;
+
+        /*
+         * for a tranpose of plain to 8c case, nodes would be like:
+         *     n    is   os
+         *     m    1    8
+         *     8    m    1
+         * or
+         *     8    m    1
+         *     m    1    8
+         */
+        ok = true
+                && (utils::one_of(a, /*4,*/ 8, 16)
+                        || utils::one_of(d, /*4,*/ 8, 16))
+                && ((b == 1 && f == 1 && a == e && c == d)
+                        || (c == 1 && e == 1 && a == f && b == d));
+        if (!ok) return false;
+
+        // Do not handle transpose of dimensions other than last 2
+        for (int i = 2; i < p.ndims; ++i) {
+            if (p.nodes[i].is != p.nodes[i].os) {
+                ok = false;
+                break;
+            }
+        }
+
+        return ok;
+    }
+
+    jit_single_blk_kernel(const tr::prb_t &prb)
+        : jit_generator()
+        , prb_(prb)
+        , ker_(nullptr)
+        , ker_tail_(nullptr)
+        , itype_sz(data_type_size(prb_.itype))
+        , otype_sz(data_type_size(prb_.otype))
+        , block_sz(prb.nodes[0].n) {
+        auto input_stride
```

Comment:
I don't see preamble. Do you preserve xmm0-xmm7 on windows?

Reply:
Sorry, for misleading (not sure how I ended up saying xmm0-xmm7). On Windows x64-calling convention requires to preserve xmm6 ... xmm15.

See here: https://docs.microsoft.com/en-us/cpp/build/x64-calling-convention?view=vs-2019#callercallee-saved-registers